src/app/component/dashboard/vaucher/vaucher.component.ts

```diff
@@ -18,19 +23,23 @@ import { HelpService } from "src/app/service/help.service";
 import { Modal } from "ngx-modal";
 import { ExcelExportData } from "@progress/kendo-angular-excel-export";
 import { Router } from "@angular/router";
+import { PackLanguageService } from "src/app/service/pack-language.service";
+import { SendSmsService } from "src/app/service/send-sms.service";
 import { checkIfInputValid } from "../../../shared/utils";
 
 @Component({
   selector: "app-vaucher",
   templateUrl: "./vaucher.component.html",
-  styleUrls: ["./vaucher.component.scss"]
+  styleUrls: ["./vaucher.component.scss"],
 })
 export class VaucherComponent implements OnInit {
-  @ViewChild('vaucher') vaucher: Modal;
-  @ViewChild('grid') grid;
+  @ViewChild("vaucher") vaucher: Modal;
+  @ViewChild("grid") grid;
 
   public allPages: boolean;
   private _allData: ExcelExportData;
+  toSendEmail: boolean = false;
```

Comment:
I don't see that this variable is used anywhere to decide to sent email. Shouldn't be if(email) then sent email, and if (sms) sent SMS. SMS part is okay, it is there, but email I didn't find